platform/src/main/java/org/hillview/sketches/DyadicHistogramBuckets.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2018 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview.sketches;
+
+import org.hillview.table.api.IColumn;
+
+/**
+ * DyadicHistogramBuckets ensures that every bucket is a multiple of the leaf granularity.
+ * It is built on data that can be converted to doubles.
+ * The last bucket is right-inclusive.
+ */
+public class DyadicHistogramBuckets implements IHistogramBuckets {
+    private double minValue;
+    private double maxValue;
+    private int numOfBuckets;
+    private final double bucketSize;
+    private final double range;
+
+    private final int granularity;
+    private final int numLeaves;
+
+    private void roundMinMax() {
+        long intMin = (long)Math.floor(this.minValue);
```

Comment:
I don't think this will work for arbitrary double values for min and max.
Consider the case where min = 0.1 and max is 0.2.